src/error.rs

```diff
@@ -1,35 +1,75 @@
 //! Error types for the PeerNet library
 
-use std::fmt::Display;
+use std::error::Error;
+use thiserror::Error;
+
+pub type PeerNetResult<T> = Result<T, PeerNetErrorData>;
 
-//TODO: Increase consistency of error handling this structure has been created just to have a place to put the errors but not designed yet.
-/// Error types for the PeerNet library
 #[derive(Debug)]
 pub enum PeerNetError {
-    /// Error when trying to start/stop a listener
-    ListenerError(String),
-    /// PeerId error
-    PeerIdError(String),
-    /// Error when trying to connect to a peer with a configuration that don't match the transport type
+    ListenerError,
+    PeerIdError,
     WrongConfigType,
-    /// Error when trying to connect to a peer
-    PeerConnectionError(String),
-    /// Error when trying to send a message to a peer
-    SendError(String),
-    /// Error when trying to receive a message from a peer
-    ReceiveError(String),
-    /// Error when trying to perform a handshake with a peer
-    HandshakeError(String),
-    /// Error when communicating with an handler
-    HandlerError(String),
-    /// Error when signing a message
-    SignError(String),
-    /// Error invalid message
-    InvalidMessage,
+    PeerConnectionError,
+    SendError,
+    ReceiveError,
+    HandshakeError,
+    HandlerError,
+    SignError,
+    SocketError,
+    BoundReached,
+
+    // Errors for specific transport types
+    TcpError,
```

Comment:
This creates an other location where we will need to update when we add a new transport that is not in the transport folder. Could we think on a more generic way maybe have a `TransportError` defined in transport/error.rs that is an enum or a trait with everything needed ?

Reply:
Yes that's what I did before, but it didn't seem like an issue for me, will modify this